File: java/src/main/java/com/myriad/wdl/model/sections/WdlInput.java
package com.myriad.wdl.model.sections;

import com.myriad.wdl.model.definitions.WdlTask.WdlTaskElement;
import com.myriad.wdl.model.definitions.WdlWorkflow.WdlWorkflowElement;
import com.myriad.wdl.model.statements.WdlDeclaration;
import java.util.ArrayDeque;

/**
 * Input section node.
 *
 * <p>This models the explicit `input { ... }` section that may appear in tasks and workflows.
 */
public final class WdlInput implements WdlTaskElement, WdlWorkflowElement {
  private final ArrayDeque<WdlDeclaration> elements = new ArrayDeque<>();

  /** Returns the ordered declarations contained in the input section. */
  public ArrayDeque<WdlDeclaration> elements() {
    return elements;
  }
}

File: typescript/src/model/sections/wdl-input.ts
/** Input section nodes. */
import type { WdlTaskElement } from '../definitions/wdl-task.js';
import type { WdlWorkflowElement } from '../definitions/wdl-workflow.js';
import type { WdlDeclaration } from '../statements/wdl-declaration.js';

/** Models an explicit `input { ... }` section in a task or workflow. */
export class WdlInput implements WdlTaskElement, WdlWorkflowElement {
  private readonly elementValues: WdlDeclaration[] = [];
  /** Returns the ordered declarations contained in the input section. */
  public elements(): WdlDeclaration[] {
    return this.elementValues;
  }
}
